src/smartcontracts/resultsParser.ts

```diff
@@ -89,12 +89,17 @@ export class ResultsParser {
     }
 
     parseOutcome(transaction: ITransactionOnNetwork, endpoint: { output: IParameterDefinition[] }): TypedOutcomeBundle {
-        let untypedBundle = this.parseUntypedOutcome(transaction);
-        let values = this.argsSerializer.buffersToValues(untypedBundle.values, endpoint.output);
+        const untypedBundle = this.parseUntypedOutcome(transaction);
+        const typedBundle = this.parseOutcomeFromUntypedBundle(untypedBundle, endpoint);
+        return typedBundle;
+    }
+
+    parseOutcomeFromUntypedBundle(bundle: UntypedOutcomeBundle, endpoint: { output: IParameterDefinition[] }) {
```

Comment:
Extracted / refactored, to be called from the transaction converter - that is, until we have the return data coming from the APIs (soon :crossed_fingers:).